src/main/java/seedu/address/model/iteration/UniqueIterationList.java

```diff
@@ -0,0 +1,142 @@
+import javafx.collections.ObservableList;
+import seedu.address.model.commission.exceptions.DuplicateCommissionException;
+import seedu.address.model.iteration.exceptions.DuplicateIterationException;
+import seedu.address.model.iteration.exceptions.IterationNotFoundException;
+
+/**
+ * A list of iterations that enforces uniqueness between its elements and does not allow nulls.
+ * An iteration is considered unique by comparing using {@code Iteration#isSameIteration(Iteration)}. As such, adding
+ * and updating of iterations uses Iteration#isSameIteration(Iteration) for equality so as to ensure that the
+ * iteration being added or updated is unique in terms of identity in the UniqueIterationList. However, the removal of
+ * an iteration uses Iteration#equals(Object) so as to ensure that the iteration with exactly the same fields will be
+ * removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Iteration#isSameIteration(Iteration)
+ */
+public class UniqueIterationList implements Iterable<Iteration> {
+
+    private final ObservableList<Iteration> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Iteration> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent iteration as the given argument.
+     */
+    public boolean contains(Iteration toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameIteration);
+    }
+
+    /**
+     * Adds an iteration to the list.
+     * The iteration must not already exist in the list.
+     */
+    public void add(Iteration toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateIterationException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the iteration {@code target} in the list with {@code editedIteration}.
+     * {@code target} must exist in the list.
+     * The iteration identity of {@code editedIteration} must not be the same as another existing
+     * iteration in the list.
+     */
+    public void setIteration(Iteration target, Iteration editedIteration) {
+        requireAllNonNull(target, editedIteration);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new IterationNotFoundException();
+        }
+
+        if (!target.isSameIteration(editedIteration) && contains(editedIteration)) {
+            throw new DuplicateIterationException();
+        }
+
+        internalList.set(index, editedIteration);
+    }
+
+    /**
+     * Removes the equivalent iteration from the list.
+     * The iteration must exist in the list.
+     */
+    public void remove(Iteration toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new IterationNotFoundException();
+        }
+    }
```

Comment:
Just wanna check can we remove by index instead since we are using list here, more efficient?

Reply:
Hm, that's a good point- I followed the implementation used by `delcus` and `delcom` which both get the Object at the specified index and removes the Object rather than removing from the index directly.

Not sure why that's the case though? Anyone has any idea if there's a specific reason for this implementation? Changing it to use a `remove(int key)` method also seems to work (but not extensively tested yet).

Reply:
Ah if I'm not wrong, I think `delcus` and `delcom` removes the Object rather than the index because the index specified is the index in the `FilteredList`. So you need to get the actual `Customer` and `Commission` from the `FilteredList` that's to be deleted and then delete that Object from the respective `Unique_____List`.

But for the case of `deliter`, it doesn't have a FilteredList, so you can just delete from the index directly. Thanks @seelengxd for pointing this out!